src/OrchardCore.Modules/OrchardCore.Forms/Workflows/Activities/ValidateAntiforgeryTokenTask.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Antiforgery;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Workflows.Abstractions.Models;
+using OrchardCore.Workflows.Activities;
+using OrchardCore.Workflows.Models;
+
+namespace OrchardCore.Forms.Workflows.Activities
+{
+    public class ValidateAntiforgeryTokenTask : TaskActivity
```

Comment:
Can you make it an option in the form task instead?

Reply:
The Form task doesn't exist yet - did you want it as part of this PR?

In any case, my philosophy behind the Forms module is that although it provides form element widgets to make it easy for users to setup a form without resorting to HTML and liquid, it does not require web developers to use it. 

In other words, the set of Workflow activities and Form element widgets are complimentary, but not dependent on one another.

For example, I should be able to create my form using Liquid (not using widgets), and use a workflow to handle its submission, including validation. As part of that validation it would be nice if I could validate the antiforgery token.

Once we have the Form task activity to handle form submissions, we should definitely add a "validate antiforgery token" option.

Let me know what you think, and I'll make the changes. Also let me know if we:

* Need the Form activity as part of this PR
* Need the ability to associate a Workflow with a Form widget as part of this PR.

The latter would allow the user to associate any workflow with the form via a select list. Or perhaps the user should be able to "activate" a workflow by clicking a button, which would then automatically create a default workflow (containing the Form activity to handle validation and submission).

Reply:
Then if not on the Form task that doesn't exist. on the Http task that will handle the POST, and if it's a POST validate the antiforgery by default.

Reply:
The issue with that is that it would no longer support 3rd party services such as GitHub to trigger workflow URLs. For that reason, we have support for SAS tokens (which are required when invoking a workflow URL).

So better would be to have a `FormSubmissionController` to which the form posts. That controller would always validate the antiforgery token, and then invoke a workflow (if one is associated with the form). I suggest we do this for Beta 3.